layers/core_checks/cc_wsi.cpp

```diff
@@ -694,6 +694,20 @@ bool CoreChecks::ValidateCreateSwapchain(const VkSwapchainCreateInfoKHR &create_
                          "contains VkImageCompressionControlEXT, but imageCompressionControlSwapchain is not enabled");
     }
 
+    const auto *swapchain_counter = vku::FindStructInPNextChain<VkSwapchainCounterCreateInfoEXT>(create_info.pNext);
+    if (swapchain_counter) {
+        VkSurfaceCapabilities2EXT surface_capabilities = vku::InitStructHelper();
+        DispatchGetPhysicalDeviceSurfaceCapabilities2EXT(physical_device, create_info.surface, &surface_capabilities);
+        if ((swapchain_counter->surfaceCounters & surface_capabilities.supportedSurfaceCounters) !=
+            swapchain_counter->surfaceCounters) {
+            skip |= LogError("VUID-VkSwapchainCounterCreateInfoEXT-surfaceCounters-01244", device,
+                             create_info_loc.dot(Struct::VkSwapchainPresentModesCreateInfoEXT, Field::surfaceCounters),
+                             "is %s, but physical device supports only %s",
```

Comment:
Maybe say something like

```suggestion
                             "is %s, but calling vkGetPhysicalDeviceSurfaceCapabilities2EXT shows only %s is supported",
```